src/SecTester.Scan/Scan.cs

```diff
@@ -0,0 +1,53 @@
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using SecTester.Scan.Models;
+
+namespace SecTester.Scan;
+
+public class Scan : IDisposable
+{
+  private static readonly HashSet<ScanStatus> ActiveStatuses =

```

Comment:
The collection must be `IReadOnlyCollection`
```suggestion
  private static readonly IReadOnlyCollection<ScanStatus> ActiveStatuses =

```